doc/guide.rst

```diff
@@ -0,0 +1,179 @@
+.. _guide_label:
+
+Guide to Simpleline
+===================
+
+Simpleline is the text user interface framework completely written in Python 3 with a possibility to
+have :ref:`non-python event loop <event_loops_label>`. With exception of optional event loops
+Simpleline doesn't have almost any dependency to external libraries.
+
+This UI is simple and easy to use. It is designed to be used with line-based machines and tools
+(e.g. serial console) so that every new line is appended to the bottom of the screen.
+Printed lines are never rewritten!
+
+Basic components
+----------------
+
+For every application the following parts are always required and a developer of an application
+will work with them.
+
+* The :class:`App <simpleline.App>` static class to initialize and run application.
+* The :class:`ScreenHandler <simpleline.render.screen_handler.ScreenHandler>` static class for
+  scheduling screens.
+* The :class:`UIScreen <simpleline.render.screen.UIScreen>` based classes to create screens which
+  will form the application.
+* :ref:`Widgets <widgets_label>` to show anything on the screens.
+* :ref:`Containers <containers_label>` to position widgets on the screen.
+
+It may look difficult to work with so many components but it is simple. Look on the next section to
+see how everything fits together.
+
+How to create simple application
+--------------------------------
+
+.. currentmodule:: simpleline.render.screen
+
+Interaction with a user is necessary to have a useful UI framework. To show anything to a user the
+:class:`UIScreen` class must be used. So we will subclass
+this class to create our screen and set title of the screen::
```

Comment:
Is there a reason for the line break after subclass? (Or does it matter when the docs are generated/rendered?)

`screen and set title of the screen::` -> `screen and set a title for it:`

Reply:
Nope I just need to wrap it manually which is sometime annoying so think I skipped/missed it here.